public/docs/ts/latest/guide/testing.jade

```diff
@@ -1,102 +1,1758 @@
+  * [Test a component inside a test host component](#component-inside-test-host)
+  * [Test a routed component](#routed-component)  
+  * [Isolated tests without the TestBed](#testing-without-testbed "Testing without the TestBed")
+  * [_TestBed_ API](#testbed-api)
+  * [FAQ](#faq "Frequently asked questions")
+:marked
+  It’s a big agenda. Fortunately, you can learn a little bit at a time and put each lesson to use.
+
+  # Live examples
+  The chapter sample code is available as live examples for inspection, experiment, and download.
+  
+  * <live-example>The sample application</live-example>
+  * <live-example plnkr="1st-specs">The first spec</live-example>
+  * <live-example plnkr="app-specs">The complete application specs</live-example>
+  * <live-example plnkr="bag-specs">A grab bag of demonstration specs</live-example>
+a(href="#top").to-top Back to top
+
+.l-hr
+a#testing-101
+:marked
+  # Introduction to Angular Testing
+
+  You write tests to explore and confirm the behavior of parts of our application.
+
+  1. They **guard** against changes that break existing code (“regressions”).
+
+  1. They **clarify** what the code does both when used as intended and when faced with deviant conditions.
+
+  1. They **reveal** mistakes in design and implementation. 
+  Tests shine a harsh light on the code from many angles. 
+  When a part of the application seems hard to test, the root cause is often a design flaw, 
+  something to cure now rather than later when it becomes expensive to fix.
+
+  This chapter assumes that you know something about testing. Don't worry if you don't. 
+  There are plenty of books and online resources to get up to speed.
+
+  <!-- TODO
+  :marked
+  ## Learn more
+  Learn more about basic Jasmine testing here
+  [Resources TBD](./#)
+  -->
+
+
+  ## Tools and Technologies
+
+  You can write and run Angular tests with a variety of tools and technologies. 
+  This chapter describes specific choices that are known to work well.
+  
+table(width="100%")
+  col(width="20%")
+  col(width="80%") 
+  tr
+    th Technology
+    th Purpose
+  tr(style=top)
+    td(style="vertical-align: top") Jasmine
+    td
+      :marked
+        The [Jasmine test framework](http://jasmine.github.io/2.4/introduction.html).
+        provides everything needed to write basic tests. 
+        It ships with an HTML test runner that executes tests in the browser.
+  tr(style=top)
+    td(style="vertical-align: top") Angular TestBed
+    td
+      :marked
+        The Angular TestBed creates an Angular test environment and harness
+        for the application code under test. 
+        Use it to condition and control parts of the application as they 
+        interact _within_ the Angular environment.
+  tr(style=top)
+    td(style="vertical-align: top") Karma
+    td
+      :marked
+        The [karma test runner](https://karma-runner.github.io/1.0/index.html)
+        is ideal for writing and running tests while developing the application. 
```

Comment:
bit of a nitpick here, but the "ideal for *writing*" sounds like karma is for writing tests apart from running them.